botocore/credentials.py

```diff
@@ -597,6 +599,62 @@ def _is_expired(self, credentials):
         return seconds < self._expiry_window_seconds
 
 
+class SessionTokenCredentialFetcher(CachedCredentialFetcher):
```

Comment:
There are few methods similar to https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L600, not sure if we want to combine them or implement base class for them.